File: lab2/Приоритетная.py
#для создания приоритетной очереди на python использоваться 2 алгоритма

# 1 вариант
from queue import PriorityQueue
# Создать приоритетную очередь
q = PriorityQueue()
# Добавить элементы в очередь с приоритетом
q.put((2, 'mid-priority item'))
q.put((1, 'high-priority item'))
q.put((3, 'low-priority item'))
# Удалить и вывести элементы в порядке приоритета
while not q.empty():
 item = q.get()
 print(item)

# 2 вариант
import heapq
customers =
heapq.heappush(customers, (2, "Harry"))
heapq.heappush(customers, (3, "Charles"))
heapq.heappush(customers, (1, "Riya"))
heapq.heappush(customers, (4, "Stacy"))
while customers:
 print(heapq.heappop(customers))

File: lab2/Приоритетная.cpp
#include <iostream>
#include <queue>
using namespace std;

int main() {
    // Приоритетная очередь целых чисел (по умолчанию сортировка по убыванию)
    priority_queue<int> pq;

    // Добавление элементов
    pq.push(10);
    pq.push(5);
    pq.push(8);
    pq.push(3);

    // Извлечение элементов (элемент с наибольшим значением будет извлечён первым)
    while (!pq.empty()) {
        cout << pq.top() << " ";  // Вывод: 10 8 5 3
        pq.pop();
    }

    return 0;
}